### ISSUE TITLE
Expected type in thrown() should be a descendant of Throwable

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sampig/spock#138]
```
The following fails with "Expected exception java.lang.String, but got 
java.util.EmptyStackException"

It would be better if expected type in thrown() is checked to be a descendant 
of Throwable and if not, it can errored out better.

---------------------------------------------
import spock.lang.*

class Spock1 extends Specification {
    def "test1"() {
       setup:
           def stack = new Stack()
        when:
          stack.pop()
        then:
          String ex = thrown()
    }
}
---------------------------------------------
```

Original issue reported on code.google.com by `roshanda...@gmail.com` on 27 Oct 2010 at 3:48

**Comment by GoogleCodeExporter** [comment]
Original comment by `pnied...@gmail.com` on 27 Oct 2010 at 11:06
- Changed state: **Fixed**
- Added labels: **Milestone-0.5**

**Comment by GoogleCodeExporter** [comment]
```
Wouldn't it be fixed on the groovy 1.7 branch as well? That's what I have 
started using locally.
```

Original comment by `roshanda...@gmail.com` on 28 Oct 2010 at 1:52

**Comment by GoogleCodeExporter** [comment]
```
I don't always merge immediately, but it's done now.
```

Original comment by `pnied...@gmail.com` on 28 Oct 2010 at 5:16
